homeassistant/components/recorder/__init__.py

```diff
@@ -656,6 +681,28 @@ def async_initialize(self) -> None:
         self._queue_watcher = async_track_time_interval(
             self.hass, self._async_check_queue, timedelta(minutes=10)
         )
+        self._keep_alive_listener = async_track_time_interval(
+            self.hass, self._async_keep_alive, timedelta(seconds=KEEPALIVE_TIME)
+        )
+        self._commit_listener = async_track_time_interval(
+            self.hass, self._async_commit, timedelta(seconds=self.commit_interval)
+        )
+
+    @callback
+    def _async_keep_alive(self, now: datetime) -> None:
+        """Queue a keep alive."""
+        if self.async_recorder_ready.is_set():
+            self.queue.put(KeepAliveTask())
+
+    @callback
+    def _async_commit(self, now: datetime) -> None:
+        """Queue a commit."""
+        if (
+            self.commit_interval
+            and not self._database_lock_task
+            and self.async_recorder_ready.is_set()
```

Comment:
We should check if there is work to commit before we insert the task. We will have to do it again in the other thread but it's a cheap check compared to the wake-up cost of the recorder thread if we don't have anything to do